### PR TITLE
[Pallas] Support multiple outputs

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request support Pallas kernels that output multiple results. The current implementation is to support an array of outputs and then do in-place updates to them. However, this somehow breaks dynamo. I will fix the dynamo issue later.

Test Plan:
PJRT_DEVICE=TPU python test/test_pallas.py -v -k test_multiple_returns

### CODE REVIEW

**Review by lsy323 — APPROVED** [review]

File: test/test_pallas.py [hunk 1]
```diff
@@ -255,6 +255,33 @@ def test_flash_attention_wrapper_bf16(self):
...
+      out_shape = jax.ShapeDtypeStruct(x.shape, x.dtype)
+      return pl.pallas_call(
+          add_minus_vectors_kernel, out_shape=[out_shape, out_shape])(x, y)
+
+    from torch_xla.experimental.custom_kernel import make_kernel_from_pallas
+    pt_kernel = make_kernel_from_pallas(
+        add_minus_vectors, lambda x, y: [(x.shape, x.dtype),
+                                         (x.shape, x.dtype)])
+    x = torch.arange(8, device="xla", dtype=torch.float)
+    o = pt_kernel(x, x)
```
Comment (JackCaoG) [inline test/test_pallas.py:278]: nit assert `len(o) == 2`

File: torch_xla/experimental/custom_kernel.py [hunk 2]
```diff
@@ -124,10 +124,17 @@ def wrapped_kernel(kernel: Callable,
         kernel, static_argnames=static_argnames).lower(*jax_args,
                                                        **kwargs).compiler_ir()
     payload = _extract_backend_config(ir)
-    output_shape, output_dtype = output_shape_dtype_fn(*args)
-    output = torch.empty(output_shape, dtype=output_dtype).to(xm.xla_device())
-    torch_xla._XLAC._xla_tpu_custom_call_(output, args, payload)
-    return output
+    # TODO: We can consider supporting un-array output.
+    outputs = []
+    for output_shape, output_dtype in output_shape_dtype_fn(*args):
```
Comment (JackCaoG) [inline torch_xla/experimental/custom_kernel.py:129]: do we need to check `output_shape_dtype_fn ` returns an array of tuple of this works for a single tuple as well?

**Review by JackCaoG — APPROVED** [review]
mostly lgtm, minor nits

**Comment by alanwaketan** [comment]
Thanks Siyuan and Jack.

Fixed all the comments.

**Pull request merged**